cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/converter/MetricConverter.java

```diff
@@ -0,0 +1,43 @@
+package com.sap.cloud.cf.monitoring.java.converter;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.sap.cloud.cf.monitoring.client.model.Metric;
+
+public abstract class MetricConverter<T> {
+    private static final String KEY_TYPE = "type";
+
+    enum MetricType {
+                     TIMER("timer"), HISTOGRAM("histogram"), GAUGE("gauge"), METER("meter"), COUNTER("counter");
+
+        private final String metricTypeName;
+
+        String getMetricTypeName() {
+            return metricTypeName;
+        }
+
+        private MetricType(String metricTypeName) {
+            this.metricTypeName = metricTypeName;
+        }
+    }
+
+    public List<Metric> convert(Map<String, T> metrics, long timestamp) {
+        ArrayList<Metric> result = new ArrayList<>();
+        for (Entry<String, T> entry: metrics.entrySet()) {
+            result.addAll(convertMetricEntry(entry, timestamp));
+        }
+        return result;
+    }
+
+    abstract List<Metric> convertMetricEntry(Entry<String, T> metricEntry, long timestamp);

```

Comment:
The default access modifier is more restrictive. All class that extends this are in the same package

Reply:
Common practice is to have abstract methods as protected. Otherwise subclasses can only be implemented in the same package, since they cannot "access" this method, while all other classes of this package can access the method.

Reply:
done